### ISSUE TITLE
Saving global styles: It is unclear what the number next to the checkbox is

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#30232]
## Description

When saving global style changes a checkbox is displayed, and next to it, there is a number. The number has no context.

Without looking through all of the code I will assume that this is a unique ID for the global styles and the active theme,
but for most users, this number is not relevant, and it is confusing both to sighted users and screen reader users.

![The save panel displays a heading and below that, a checkbox with the label "5" ](https://user-images.githubusercontent.com/7422055/112455640-332b9780-8d5a-11eb-9635-7d5243761b00.png)

With a screen reader, the panel reads like this:
Save  button  
Close panel  button  
Global Styles  button  expanded    heading    level 2
5  check box  checked

## Step-by-step reproduction instructions

1. Activate a theme with support for full site editing and open the site editor.
2. Make changes to global styles, and save.
3. View the number in the save panel.

## Expected behaviour

I expect the label to be relevant and describe what I am saving.

## Actual behaviour

A number is displayed, and it has no context.

## WordPress information
- WordPress version: 5.7
- Gutenberg version: current trunk at https://github.com/WordPress/gutenberg/commit/6be973b4d435aa5458e0731f5306d1588e5c7c98
- Are all plugins except Gutenberg deactivated?  yes
- Are you using a default theme (e.g. Twenty Twenty-One)?  TT1 Blocks, Empty theme

## Device information
- Device: Desktop
- Operating system: Windows 10
- Browser: Chrome Version 89.0.4389.90

**Comment by carolinan** [comment]
I found a similar issue if there is a post in the site editor, and the post has no post title.
I did not confirm but assumed this is the post ID.

![if a post without a title is edited in the site editor, the post id(?) is showing](https://user-images.githubusercontent.com/7422055/112460367-19408380-8d5f-11eb-9897-cded4a1f3b5e.png)

**Comment by mtias** [comment]
Yes, this definitely needs some improvements :)

cc @javierarce as part of looking at the multi-entity saving flow.

**Comment by mtias** [comment]
An initial thought is this should always use the "title". If a post doesn't have a title, should say "(untitled)". For global styles, it'd be nice to set a title on the post type object that holds the style info like "Custom Styles". cc @nosolosw

**Comment by oandregal** [comment]
https://github.com/WordPress/gutenberg/pull/30521 fixes this.
